site/src/pages/WorkspacePage/WorkspaceActions/Buttons.tsx

```diff
@@ -190,3 +192,24 @@ export const RetryButton: FC<RetryButtonProps> = ({
     </TopbarButton>
   );
 };
+
+interface FavoriteButtonProps {
+  handleAction: (workspaceID: string) => void;
+  workspaceID: string;
+  isFavorite: boolean;
+}
+
+export const FavoriteButton: FC<FavoriteButtonProps> = ({
+  handleAction,
+  workspaceID,
+  isFavorite,
+}) => {
+  return (
+    <TopbarButton
+      startIcon={isFavorite ? <Star /> : <StarBorder />}
+      onClick={() => handleAction(workspaceID)}
+    >
+      {isFavorite ? "Unfavorite" : "Favorite"}
```

Comment:
Since this button can have two states, favorite and unfavorite, I would create a story in a storybook for both scenarios.